### PR TITLE
[@types/commercetools__enzyme-extensions] Remove renderProp from commercetools__enzyme-extensions

### LINKED ISSUE #53412: [@types/enzyme] ReactWrapper doesn't have renderProp method, while it exists in fact
According to the[ docs](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/renderProp.html), method `renderProp` should exist on both `ShallowWrapper` and `ReactWrapper`, but in the types file it is only listed in the 'ReactWrapper` class.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2e1ef146afb5dca4c7cc3175f929adf85c39129e/types/enzyme/index.d.ts#L493

I suggest either duplicating the same method to the `ReactWrapper` or moving it on an upper level to the `CommonWrapper` interface.

### PR DESCRIPTION
**Pull request by RAYDENFilipp**
According to the [project page](https://github.com/commercetools/enzyme-extensions) support for `renderProp` was dropped since v4.0.

This is needed to create MR with [patch-2](https://github.com/RAYDENFilipp/DefinitelyTyped/tree/patch-2) to close #53412

But since @types/enzyme is coupled with @types/commercetools__enzyme-extensions when testing all libraries, I had to update @types/commercetools__enzyme-extensions first

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/commercetools/enzyme-extensions
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by rbuckton — APPROVED** [review]

**Comment by samarmohan** [comment]
@RAYDENFilipp @screendriver can someone merge this please?

**Review by samarmohan — APPROVED** [review]

**Comment by RAYDENFilipp** [comment]
@samarmohan Thanks for reviewing. 

Sure, I can merge it, but before that could you tell me, please, if everything was done correctly? Because it was my first try and I am still not sure if that's how typings are undated for a library that dropped some functionality. Thank you.

**Comment by samarmohan** [comment]
That was perfect, I don’t really know your exact workflow but it’s good, keep it up!

**Comment by RAYDENFilipp** [comment]
@samarmohan  I just overlooked that I actually don't have access rights, so it should be someone else who does. Thanks for your answer, have a great weekend!

**Comment by samarmohan** [comment]
Oh, you can’t merge because of some technicality, wait for a maintainer to approve than you can merge.

**Comment by samarmohan** [comment]
Dang it, @RAYDENFilipp can you remake this pr (because it is needed) or if you don’t want to I can.

**Comment by RAYDENFilipp** [comment]
@samarmohan  Sure

**Comment by RAYDENFilipp** [comment]
@samarmohan https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54622

**Pull request closed without merging**
